docs/API.md

```diff
@@ -96,6 +96,19 @@ __Parameters__
 |`ssl` | _bool_  | If 'true' API requests will be secure (HTTPS), and insecure (HTTP) otherwise |
 |`region`| _string_ | Region for the object storage |
 
+### NewWithVirtual(endpoint, accessKeyID, secretAccessKey string, ssl bool, region string) (*Client, error)
+Initializes minio client. Use this constructor if server supports virtual hosted style requests. By default path style requests are used unless endpoint is Amazon/Google cloud storage.
+
+__Parameters__
+
+|Param   |Type   |Description   |
+|:---|:---| :---|
+|`endpoint`   | _string_  |S3 compatible object storage endpoint |
+|`accessKeyID`  |_string_   |Access key for the object storage |
+|`secretAccessKey`  | _string_  |Secret key for the object storage |
+|`ssl` | _bool_  | If 'true' API requests will be secure (HTTPS), and insecure (HTTP) otherwise |
+|`region`| _string_ | Region for the object storage |
```

Comment:
Missing virtual..

Reply:
deliberately so - since the default is path style in any case, using this constructor indicates user wants to explicitly use virtual hosted style. If a bool virtual flag was permitted, the  SDK would need to maintain 3 states internally to be able to identify whether user specified virtual=false for endpoints such as Amazon where we default to virtual style. It wouldn't add much value to provide an additional parameter in the constructor, when user could instantiate with New/NewWithRegion when path style is desired.

Reply:
I think this is time to make our New a much more future compatible and break the version compatiblity.

Reply:
meaning, we provide a virtual flag in the constructor and remove default behavior for aws/google cloud?

Reply:
My proposal is we bring PutObjectOptions like behavior for our New()

```go
NewWithOptions(endpoint, opts *Options)

type Options struct {
       Creds *credentials.Credentials
       Secure bool
       Region string
       VirtualStyle bool
       // Add future fields here
}
```

Reply:
yes, it looks cleaner. will check with @abperiasamy